mlrun/execution.py

```diff
@@ -93,6 +94,7 @@ def __init__(self, autocommit=False, tmp="", log_stream=None):
         self._outputs = []
 
         self._results = {}
+        # tracks the execution state, completion of runs is decided by the API
```

Comment:
do you mean terminal state? need to have in mind the other terminal states

Reply:
As I said in the other comments. I think the only terminal state that needs special attention here is `completion` but this is open for discussion